### PR TITLE
AuthenticationHelper.js has more unit test coverage

### PR DESCRIPTION
**Pull request by evcodes**
#### Description of how you validated changes
After creating the various unit tests, I ran ```yarn test``` to ensure that the tests would catch nearly every branch, happy case, and edge case.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/Utils.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//promisify is a utility function to clean up the repetition
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/Utils.js:1]: I did have it there but unfortunately Jest tries to run test across all files in the __test__ folder

File: packages/amazon-cognito-identity-js/src/AuthenticationHelper.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ import WordArray from './utils/WordArray';
  * @param {number} nBytes 
  * @returns {Buffer} fixed-length sequence of random bytes
  */
-function randomBytes(nBytes) {
+export function randomBytes(nBytes) {
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/src/AuthenticationHelper.js:31]: I am using this function within the AuthenticationHelper file

File: packages/amazon-cognito-identity-js/Utils.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+//promisify is a utility function to clean up the repetition 
+// in jest's asynchronous done() pattern
+// More information here:  https://jestjs.io/docs/asynchronous#callbacks
+
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/Utils.js:4]: Will do!

File: packages/amazon-cognito-identity-js/Utils.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	return new Promise((resolve, reject) => {
+		obj[fn].apply(obj, [...args, (err, data) => {
+			if (err) {
+				reject(err);
+			}
+			else {
+				resolve(data);
+			}
+		}]);
+	});
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/Utils.js:15]: I changed it to the more slick

```
return new Promise((resolve, reject) => {
		let callback = (err, data) => { err ? reject(err) : resolve(data) }
		obj[fn].apply(obj, [...args, callback]);
	});
```

File: packages/amazon-cognito-identity-js/Utils.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+
+// instance.k = new BigInteger('deadbeef', 16)
+// 		instance.UValue = instance.calculateU(instance.largeAValue, xValue)
+// 		const xValue = new BigInteger('deadbeef', 16)
+// 		const serverValue = new BigInteger('deadbeef', 16)
+// 		const result = await promisify(instance, 'calculateS', xValue, serverValue)
+//		typeof result === Error ? expect(result).toMatchObject(new Error()) : expect(result).toBe(null)
+
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/Utils.js:28]: nope this is trash, sorry for my oversight

File: packages/amazon-cognito-identity-js/__tests__/AuthenticationHelper.test.js [hunk 6]
```diff
@@ -553,4 +555,145 @@ describe('AuthenticatorHelper', () => {
...
+		await promisify(instance, 'getPasswordAuthenticationKey', username, password, realServerValue, salt).catch(e => {
+			expect(e).toEqual(new Error('U cannot be zero.'))
+		})
+	})
+
+	test('Getting the password auth key', async () => {
+		const realServerValue = new BigInteger('deadbeef', 16)
+		const result = await promisify(instance, 'getPasswordAuthenticationKey', username, password, realServerValue, salt)
+
+		expect(result).toBeInstanceOf(Buffer)
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/__tests__/AuthenticationHelper.test.js:696]: I don't think so since it depends on the hkdf key derivation algorithm. I was told by Manuel and the team to treat these functions as black boxes. I also wasn't very sure how to test this function explicity.

File: packages/amazon-cognito-identity-js/__tests__/AuthenticationHelper.test.js [hunk 7]
```diff
@@ -553,4 +555,145 @@ describe('AuthenticatorHelper', () => {
...
+		expect(instance.getVerifierDevices()).toBe(instance.verifierDevices)
+	})
+
+	test('Constant prefix for new password challenge', () => {
+		expect(instance.getNewPasswordRequiredChallengeUserAttributePrefix()).toEqual('userAttributes.')
+	})
+
+	test('Get large A value', async () => {
+		const result = await promisify(instance, 'getLargeAValue')
+		expect(result).toBe(instance.largeAValue)
```
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/__tests__/AuthenticationHelper.test.js:582]: actually the getLargeAValues is a calculation and we want to ensure that the calculation (calculateA) gets the right result.

File: packages/amazon-cognito-identity-js/src/AuthenticationHelper.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (evcodes) [reply packages/amazon-cognito-identity-js/src/AuthenticationHelper.js:31]: correction: I thought I was using this but I am not anymore and therefore, I will remove it.
